erdera/mapping_cnag_to_rd3.py

```diff
@@ -21,88 +22,130 @@ def get_staging_area_participants():
+        others_affected = pedigree.loc[pedigree['id'] == family_id, 'others affected'].to_list()
+        non_na = [x for x in others_affected if not np.isnan(x)] # only keep non NA values
+        non_na_unique = set(non_na) # get the unique values
+        if len(non_na_unique) == 1: # set field
+            pedigree.loc[pedigree['id'] == family_id, 'others affected'] = next(iter(non_na_unique))
+        # if the length is more than 1, this field for this value is both True and False, 
+        # this cannot be the case so print a message to the user 
+        elif len(non_na_unique) > 1: 
+            print(f"Inconsistent values found for 'others affected' for family {family_id}: {non_na_unique}")
+        
+    pedigree = pedigree.drop_duplicates()
+            
     # upload
     client.save_schema(table='Pedigree', data=pedigree)
 
+    # work in progress - unfinished
+    # updated_records = pedigree.loc[pedigree['is new'] == False].copy() # get the records to be updated
+    current_pedigrees = client.get(table='Pedigree', as_df=True)
+
+    current_pedigrees_ids = current_pedigrees['id']
+    pedigree['is new pedigree'] = ~pedigree['id'].isin(current_pedigrees_ids)
+
+    merged = pedigree.merge(current_pedigrees, on='id', how='inner', suffixes = ('_new', '_old'))
+
+    changed_mask = (merged
+                    .filter(like='_new')
+                    .ne(merged.filter(like='_old').values)
+                    .any(axis=1)
+                    )
+    
+    merged['changed'] = changed_mask
+
 def build_import_individuals_table(client, data: pd.DataFrame):
     """Map staging area data into the Individuals table"""
     individuals = data[['id', 'sex', 'lifeStatus', 'report_date', 'last_modification_date', 'report_id', 'baselineage']] \
-    .rename(columns={
-        'id': 'alternate ids', 
-        'sex': 'gender at birth',
-        'lifeStatus': 'individual status', 
-        'report_date': 'date created at source', 
-        'last_modification_date': 'date updated at source', 
-        'report_id': 'id',
-        'baselineage': 'age at enrolment'
-    })
+        .rename(columns={
+            'id': 'alternate ids',
+            'sex': 'gender at birth',
+            'lifeStatus': 'individual status',
+            'report_date': 'date created at source',
+            'last_modification_date': 'date updated at source',
+            'report_id': 'id',
+            'baselineage': 'age at enrolment'
+        })
 
     individual_status_dict = {
         'Deceased': 'Dead'
     }
-    individuals['individual status'] = individuals['individual status'].replace(individual_status_dict)
-    
+    individuals['individual status'] = individuals['individual status'].replace(
+        individual_status_dict)
+
     # map sex
     gender_dict = {
-        'M': 'assigned male at birth', 
+        'M': 'assigned male at birth',
         'F': 'assigned female at birth'
     }

```

Comment:
For a future PR: We could also put these in the ontology mapping schema. I'm not sure if we will ever receive other values, but it might be easier to manage the mappings and new values.